src/codegen/c.zig

```diff
@@ -308,6 +308,10 @@ pub const DeclGen = struct {
                 if (ty.isPtrLikeOptional()) {
                     return dg.renderValue(writer, payload_type, val);
                 }
+                if (payload_type.abiSize(builtin.target) == 0) {
```

Comment:
Be sure to pass in the target that is chosen to build for, not the host target that the compiler is running on! Look for `const target = ` elsewhere in this file. I see other instances of `builtin.target` in this patch too.